pkg/operators/otel-logs/otel-logs.go

```diff
@@ -0,0 +1,320 @@
+		o.providers[k] = provider
+		log.Debugf("> log exporter %q with endpoint %q loaded", k, v.Endpoint)
+	}
+
+	return nil
+}
+
+func (o *otelLogsOperator) GlobalParams() api.Params {
+	return api.Params{}
+}
+
+func (o *otelLogsOperator) InstanceParams() api.Params {
+	return api.Params{
+		&api.Param{
+			Key:          ParamOtelLogsExporter,
+			Description:  "Exporter to use for log exporting",
+			DefaultValue: "",
+		},
+	}
+}
+
+func (o *otelLogsOperator) InstantiateDataOperator(gadgetCtx operators.GadgetContext, instanceParamValues api.ParamValues) (operators.DataOperatorInstance, error) {
+	if len(o.providers) == 0 {
+		return nil, nil
+	}
+	mappings, err := apihelpers.GetStringValuesPerDataSource(instanceParamValues[ParamOtelLogsExporter])
+	if err != nil {
+		return nil, fmt.Errorf("parsing name mappings: %w", err)
+	}
+	inst := &otelLogsOperatorInstance{
+		o:        o,
+		mappings: mappings,
+		loggers:  make(map[datasource.DataSource]otellog.Logger),
+	}
+	err = inst.init(gadgetCtx)
+	if err != nil {
+		return nil, err
+	}
+	return inst, nil
+}
+
+func (o *otelLogsOperator) Priority() int {
+	return 9999
+}
+
+type otelLogsOperatorInstance struct {
+	o        *otelLogsOperator
+	mappings map[string]string
+	loggers  map[datasource.DataSource]otellog.Logger
+}
+
+func (o *otelLogsOperatorInstance) init(gadgetCtx operators.GadgetContext) error {
+	for _, ds := range gadgetCtx.GetDataSources() {
+		annotations := ds.Annotations()
+
+		// Find mapping
+		exporterName, ok := o.mappings[ds.Name()]
+		if !ok {
+			exporterName, ok = o.mappings[""]
+			if !ok {
+				continue
+			}
+		}
+
+		exporter, ok := o.o.providers[exporterName]
+		if !ok {
+			return fmt.Errorf("exporter not found: %q", exporterName)
+		}
+
+		loggerName := annotations[AnnotationLogsName]
+		if loggerName == "" {
+			loggerName = gadgetCtx.ImageName()
+		}
+
+		gadgetCtx.Logger().Debugf("logging %q to exporter %q", ds.Name(), exporterName)
+		o.loggers[ds] = exporter.Logger(loggerName)
+	}
+	return nil
+}
+
+func (o *otelLogsOperatorInstance) Name() string {
+	return "otel-logs"
+}
+
+func (o *otelLogsOperatorInstance) PreStart(gadgetCtx operators.GadgetContext) error {
+	for ds, logger := range o.loggers {
+		fields := ds.Accessors(false)
+		annotations := ds.Annotations()
+
+		prep := make([]func(data datasource.Data) otellog.KeyValue, 0)
+		kvCount := 0
+
+		fns := make([]func(datasource.Data, *otellog.Record), 0)
+
+		// fixed severity by annotation
+		if severity, ok := annotations[AnnotationLogsSeverity]; ok {
+			sv, err := strconv.ParseFloat(severity, 64)
+			if err != nil {
+				return fmt.Errorf("invalid log severity %q: %w", severity, err)
+			}
+			fns = append(fns, func(data datasource.Data, record *otellog.Record) {
+				record.SetSeverity(otellog.Severity(sv))
+			})
+		}
+
+		// body string by annotation
+		if bodyString, ok := annotations[AnnotationLogsBody]; ok {
```

Comment:
Should we return an error if we aren't able to find `bodyString` ? or is there a default body format if nothing is specified?

Reply:
If the body is empty, it will still log the fields and such; the body in otel would in that case show as INVALID. I think it's fine for now.